File: recursion.java
import java.util.Scanner;

public class Main {
    public static boolean isPalindrome(String s, int left, int right) {
        if (left >= right) return true;
        if (s.charAt(left) != s.charAt(right)) return false;
        return isPalindrome(s, left + 1, right - 1);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Введите строку: ");
        String input = sc.nextLine();
        
        if (isPalindrome(input, 0, input.length() - 1))
            System.out.println("Палиндром");
        else 
            System.out.println("Не палиндром");
    }
}

// Пример ввода:

// level

// Вывод:

// Палиндром

File: recursion.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindrome(const string &s, int left, int right) {
    if (left >= right) return true;
    if (s[left] != s[right]) return false;
    return isPalindrome(s, left + 1, right - 1);
}

int main() {
    string s;
    cout << "Введите строку: ";
    getline(cin, s);

    if (isPalindrome(s, 0, s.size() - 1))
        cout << "Палиндром";
    else
        cout << "Не палиндром";
}

// Пример ввода:

// madam

// Вывод:

// Палиндром
